pkg/build/api/validation/validation_test.go

```diff
@@ -116,8 +150,19 @@ func TestBuildConfigValidationFailure(t *testing.T) {
 			},
 		},
 	}
-	if result := ValidateBuildConfig(buildConfig); len(result) != 2 {
-		t.Errorf("Unexpected validation result %v", result)
+	errors := ValidateBuildConfig(buildConfig)
+	if len(errors) != 1 {
+		t.Fatalf("Unexpected validation errors %v", errors)
+	}
+	err := errors[0].(*fielderrors.ValidationError)
+	if err.Type != fielderrors.ValidationErrorTypeInvalid {
+		t.Errorf("Unexpected error type, expected %s, got %s", fielderrors.ValidationErrorTypeInvalid, err.Type)
+	}
+	if err.Field != "triggers" {
+		t.Errorf("Unexpected field name expected name, got %s", err.Field)
+	}
+	if !strings.Contains(err.Detail, "only one ImageChangeTrigger is allowed") {
+		t.Errorf("Unexpected error details: %s", err.Detail)
```

Comment:
constants become a pain since now you've moved the message away from the primary place you're using the message so you have to refer back and forth.

in anycase i think we're bikeshedding at this point.